sources/cqlrt_common.c

```diff
@@ -5218,7 +5202,7 @@ static uint32_t cql_read_big_endian_u32(const uint8_t *_Nonnull b) {
   val = (val << 8) | b[1];
   val = (val << 8) | b[2];
   val = (val << 8) | b[3];
-  return val;
+  return (cql_uint32)val;
```

Comment:
better to change val into cql_uint32 a few lines up.  It has no reason to be uint64_t